File: sdks/local-rp/java/src/test/java/community/catalyst/linkkeys/localrp/TicketsConformanceTest.java
package community.catalyst.linkkeys.localrp;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;

import java.util.List;

import org.junit.jupiter.api.Test;

import community.catalyst.linkkeys.localrp.crypto.Crypto;
import community.catalyst.linkkeys.localrp.testutil.Fixtures;
import community.catalyst.linkkeys.localrp.testutil.MiniJson.JsonValue;

/** Conformance vectors: {@code tickets.json}. */
class TicketsConformanceTest {

    @Test
    void hashPairsMatchFingerprintRoutine() {
        JsonValue d = Fixtures.load("tickets.json");
        List<JsonValue> cases = d.get("cases").asArray();
        assertFalse(cases.isEmpty());
        for (JsonValue c : cases) {
            byte[] ticket = Fixtures.hex(c.get("ticket_hex").asString());
            assertEquals(32, ticket.length);
            assertEquals(c.get("sha256_hex").asString(), Crypto.fingerprint(ticket));
        }
    }
}

File: sdks/local-rp/kotlin/src/test/kotlin/community/catalyst/linkkeys/localrp/kt/TicketsConformanceTest.kt
package community.catalyst.linkkeys.localrp.kt

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Test
import community.catalyst.linkkeys.localrp.kt.protocol.Crypto
import community.catalyst.linkkeys.localrp.kt.testutil.Fixtures

/** Conformance vectors: `tickets.json`, exercised through this package's own [Crypto.fingerprint]. */
class TicketsConformanceTest {

    @Test
    fun hashPairsMatchFingerprintRoutine() {
        val d = Fixtures.load("tickets.json")
        val cases = d.get("cases").asArray()
        assertFalse(cases.isEmpty())
        for (c in cases) {
            val ticket = Fixtures.hex(c.get("ticket_hex").asString())
            assertEquals(32, ticket.size)
            assertEquals(c.get("sha256_hex").asString(), Crypto.fingerprint(ticket))
        }
    }
}
